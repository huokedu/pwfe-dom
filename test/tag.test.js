/**
 * Created by chkui on 2017/6/20.
 */

import React from 'react'
import Tag from '../tag'
import {HashRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
beforeAll(() => {
    Tag.setIcon({
        test:"base64:test",
        act1:"base64:act1",
        act2:"base64:cat2"
    })
});

test("Tag.A Test", ()=> {
    const comp = renderer.create(
        <HashRouter>
            <Tag.A>https://github.com/palmg/pwfe-dom</Tag.A>
        </HashRouter>
    )

    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
})

test("Tag.Img Test", ()=> {
    const comp = renderer.create(
        <Tag.Img src="file.icon"/>
    )

    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
})

test("Tag.Icon Test none", ()=> {
    const comp = renderer.create(
        <Tag.Icon/>
    )

    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
})

test("Tag.Icon Test value", ()=> {
    const comp = renderer.create(
        <Tag.Icon src="test" alt="测试图片" style={{width:'20px'}} className="css"/>
    )

    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
})

test("Tag.Icon DynaIcon", ()=> {
    const comp = renderer.create(
        <Tag.DynaIcon src="act1" actSrc="act2" alt="测试图片" style={{width:'20px'}} className="css"/>
    )

    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
})
test("Tag.Icon DynaIcon act", ()=> {
    const comp = renderer.create(
        <Tag.DynaIcon src="act1" actSrc="act2" act alt="测试图片" style={{width:'20px'}} className="css"/>
    )

    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
})