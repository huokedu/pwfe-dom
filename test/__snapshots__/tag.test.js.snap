// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag.A Test 1`] = `
<a
  onClick={[Function]}
>
  https://github.com/palmg/pwfe-dom
</a>
`;

exports[`Tag.Icon DynaIcon 1`] = `
<img
  alt="测试图片"
  className="css"
  src="base64:act1"
  style={
    Object {
      "width": "20px",
    }
  }
/>
`;

exports[`Tag.Icon DynaIcon act 1`] = `
<img
  alt="测试图片"
  className="css"
  src="base64:cat2"
  style={
    Object {
      "width": "20px",
    }
  }
/>
`;

exports[`Tag.Icon Test none 1`] = `
<img
  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMoAAADKCAYAAADkZd+oAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA4RpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuNS1jMDE0IDc5LjE1MTQ4MSwgMjAxMy8wMy8xMy0xMjowOToxNSAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo2YmI5NjAwMy0wZjdjLTRkYWEtODMxMC05YjcwYzVhYTk2YWMiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6NUU5QzgzQzY0MTI0MTFFN0JGNTFEQzI4RDQ1NTMyMDciIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6NUU5QzgzQzU0MTI0MTFFN0JGNTFEQzI4RDQ1NTMyMDciIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENDIDIwMTcgKFdpbmRvd3MpIj4gPHhtcE1NOkRlcml2ZWRGcm9tIHN0UmVmOmluc3RhbmNlSUQ9InhtcC5paWQ6ZDA5NGU2ODItM2M0Ni0xYzRhLWE4Y2UtMDcwOGVlZGRhZGNlIiBzdFJlZjpkb2N1bWVudElEPSJhZG9iZTpkb2NpZDpwaG90b3Nob3A6MzAxYjRkN2ItMmYwNi0xMWU3LWJmZjgtY2Y1NmI0OGFlYjMxIi8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+cQqqIwAAAaJJREFUeNrs0zENAAAIBDHAv+fHARNjK+GS6yQF3EYCMAoYBYwCRgGjgFHAKIBRwChgFDAKGAWMAkYBjAJGAaOAUcAoYBQwChgFMAoYBYwCRgGjgFHAKIBRwChgFDAKGAWMAkYBjAJGAaOAUcAoYBQwChgFMAoYBYwCRgGjgFHAKIBRwChgFDAKGAWMAkYBjAJGAaOAUcAoYBQwChgFMAoYBYwCRgGjgFHAKIBRwChgFDAKGAWMAkYBowBGAaOAUcAoYBQwChgFMAoYBYwCRgGjgFHAKIBRwChgFDAKGAWMAkYBowBGAaOAUcAoYBQwChgFMAoYBYwCRgGjgFHAKIBRwChgFDAKGAWMAkYBowBGAaOAUcAoYBQwChgFMAoYBYwCRgGjgFHAKGAUwChgFDAKGAWMAkYBowBGAaOAUcAoYBQwChgFMAoYBYwCRgGjgFHAKGAUwChgFDAKGAWMAkYBowBGAaOAUcAoYBQwChgFMAoYBYwCRgGjgFHAKGAUwChgFDAKGAWMAkYBowBGAaOAUcAoYBQwChgFjAIYBT6sAAMAS8gEkf6ULmwAAAAASUVORK5CYII="
/>
`;

exports[`Tag.Icon Test value 1`] = `
<img
  alt="测试图片"
  className="css"
  src="base64:test"
  style={
    Object {
      "width": "20px",
    }
  }
/>
`;

exports[`Tag.Img Test 1`] = `
<img
  alt="测试图片"
  className={undefined}
  src="file.icon"
/>
`;
